src/auth0-sso-login.js

```diff
@@ -12,48 +14,58 @@ export default class auth {
   constructor(config) {
     this.config = config || {};
     this.authResult = null;
+    let logger = new Logger(config);
+    this.logger = logger;
     this.tokenExpiryManager = new TokenExpiryManager();
+    this.redirectHandler = new RedirectHandler(logger);
     this.renewAuthSequencePromise = Promise.resolve();
   }
 
   /**
-   * @description logs the message to the console, or to a provided hook
-   * @param message to log
-   * @return {*|void}
+   * @description update the detailed profile with a call to the Auth0 Management API
+   * @param idToken the jwt for a user
+   * @return {Promise<any>} resolved promise with user profile; rejected promise with error
    */
-  log(message) {
-    const logFunc = (this.config.hooks ? this.config.hooks.log : undefined) || console.log;
-    return logFunc(message);
+  refreshProfile() {
+    return this.getProfile()
+    .then(profile => {
+      this.profileRefreshed(profile);
+    }, error => {
+      this.logger.log({ title: 'Error while retrieving user information after successful authentication', error: error });
+    });
   }
 
   /**
-   * @description gets the detailed profile with a call to the Auth0 Management API
-   * @param idToken
-   * @param sub
-   * @return {Promise<any>} resolved promise with user profile; rejected promise with error
+   * @description Get the latest available profile
+   * @return {null|Object} profile if the user was already logged in; null otherwise
    */
-  getDetailedProfile(idToken, sub) {
+  getProfile() {
+    let idToken = this.authResult.idToken;
+    let jwt = jwtManager.decode(idToken);
```

Comment:
we just got the token back from auth0, there is no way for it to be invalid.  `verify` has the unnecessary overhead of downloading the public key from `auth0.com` and then trying to validate it.
  Does it really matter if the logged in user manually replaces the `jwt`, no, because they can already do that by logging in, or changing this method on the screen.

As for the
> do we need a `try/catch` block around it

I can add one, but it seems that [jsonwebtoken](https://npm.runkit.com/jsonwebtoken) was fairly good unless I explicitly create an invalid token to try to get to crash.